pkg/trace/stats/aggregation.go

```diff
@@ -44,6 +44,11 @@ type PayloadAggregationKey struct {
 }
 
 func getStatusCode(s *pb.Span) uint32 {
+	code, ok := traceutil.GetMetric(s, tagStatusCode)
+	if ok {
+		// only 7.39.0+, for lesser versions, always use Meta
+		return uint32(code)
+	}
 	strC := traceutil.GetMetaDefault(s, tagStatusCode, "")
```

Comment:
In case both are provided, only the Metric one is considered as Status Code. Is that correct?

Reply:
That’s what I thought makes most sense, because after all, the HTTP status code *is* a numeric value!

Reply:
Note that our docs today require that `http.status_code` be sent as a string attribute (for historical reasons):
https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2357395856/Span+attributes#http.status_code

I'm not sure if we should make an integer value from `metrics` have higher priority than a string from `meta` without at least warning the tracer teams and documenting that behavior in the document above.

cc @brettlangdon

Reply:
In Ruby, it's not possible to set both meta and metrics with the same key: the last one set will win. I believe this behavior was part of the requirements when metrics was introduced.

Reply:
@lucaspimentel I'm resolving this. If you feel that there could be issues with this approach, please re-open.